provider/oci/networking.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package oci
+
+import (
+	"github.com/juju/errors"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/instance"
+	"github.com/juju/juju/network"
+)
+
+// Subnets is defined on the environs.Networking interface.
+func (e *Environ) Subnets(id instance.Id, subnets []network.Id) ([]network.SubnetInfo, error) {
```

Comment:
If you're not actually planning on implementing support for environs.Networking or environs.NetworkEnviron we could probably just leave off these functions rather than have them all implemented but returning nil.

Reply:
I am hoping that attaching extra NICs on instance spin up will be implemented in OCI in the not too distant future, at which point, this will be doable. If you think it's better to remove it, I can.